### DISCUSSION TITLE
Is GatsbyJS Officially Dead?

### DISCUSSION THREAD

**Discussion by JR-Carroll** [gatsbyjs/gatsby#39062] [Community]
I can't get Netlify to respond, and I read rumors online suggesting the original GatsbyJS team rage-quit or was let-go (whatever) Netlify post-merger -- whatever the facts, I have several GatsbyJS-based projects I need to know if I should migrate to another framework.

1. Is GatsbyJS officially dead or is it just _dead-silent_? I see 'fix' commits in the repo, but no major dev. And the blogs between GatsbyJS and Netlify are pretty silent for the last 1-1.5years.
2. For those migrating away, what's the preferred equivalent framework? NextJS? Nuxt? or?
3. What's the real story with what happened to GatsbyJS? I've LOVED Gatsby, and have built a mini-side-hustle that's done very well with GatsbyJS, but I feel abandoned and I want closure.

Thanks GatsbyJS-residual-team! I miss you guys and hope everything is OK!

**Comment by jcarroll-ceo** [comment]
Following up on this thread. I would really appreciate the maintainers perspective on GatsbyJS future. I have projects that are dependent on this and I need to advise my clients. 

Any community member thoughts on the health and survivability of GatsbyJS as it lives in the Netlify-ecosystem? 

(apologies, not trying to pretend Im another account -- Im the OP, just happened to check status while I was in my other account).

**Comment by kremalicious** [comment]
It has been abandoned and therefore can be considered dead, see the silence of maintainers on https://github.com/gatsbyjs/gatsby/issues/38696.

Next, Astro, Remix are all good alternatives.

**Comment by tiggeymone** [comment]
As far as I'm concerned, Netlify is losing a lot in terms of reputation for what it's doing with Gatsby. I can't recall another operation so deliberately harmful to the JS community as Netlify's acquisition of Gatsby. An entire team laid off overnight, a SaaS with thousands of active projects trashed, and lots of "don't worry, everything is fine" messages while the building is on fire.

I hope for more transparency from the Netlify team in the future, because there's a big difference between saying "Gatsby may no longer be cutting-edge, but we're focusing on stability" and not having a significant commit in almost a year.

I still have several projects in Gatsby, and I can't find a valid alternative when it comes to theme shadowing. It doesn't seem like Astro supports it. Do you have any valid suggestions?

**Reply by McLeodSean** [reply]
@tiggeymone - Would love to know if you come across an alternative specifically for shadowing. It seems to be a fairly niche feature, but it's been incredibly valuable for us and is pretty much the only thing stopping us from migrating elsewhere.

The only viable alternatives I've found are:

1. Rolling a custom webpack config using [webpack-theme-component-shadowing](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/internal-plugins/webpack-theme-component-shadowing/index.js) - I'd prefer to move away from Webpack
2. vite-plugin-blueprint - An obscure and unmaintained Vite plugin

**Reply by ZeldOcarina** [reply]
Astro

**Comment by JR-Carroll** [comment]
I've had it narrowed down to Astro and Next for awhile, but I am very allergic to going Astro for the very simple reason that Astro appears to be a kissing-cousins to GatsbyJS/Netlifys model. Say I run with them and they implode just like Netlify/GatsbyJS - shame on me!

**Reply by munawar-js** [reply]
We are actually considering to switch our [gatsby-starter-foundation](https://github.com/stackrole/gatsby-starter-foundation) to Astro, how sure are you about this.

**Reply by JR-Carroll** [reply]
Sure about what? Astro going the way of Gatsby? Im not certain of that at all. My comment is entirely fear based given the limited research I've done on Astro.

Im not interested in good-willed private enterprises that play the cash-grab game. I'm just looking for a stable framework to deploy to a stable platform. It's going to cost me a lot of time, which is money, to move my projects away from Gatsby. I accept that things like this happen, but the way GatsbyJS/Netlify handled this really doesn't sit right with me and it feels like Im prematurely migrating away from Gatsby when it could have been a viable option for many years to come.

For now, I'm personally picking NextJS. Yes... privately owned by Vercel, but my research suggest it's much more versatile than Astro (this all comes from anecdotes off of Reddit -- ask me again in a few months if I made the right choice). The other thing that makes me raise an eyebrow is that Astro has some component architecture that sounds like it may not mix well. AKA going Astro might be a one-way trip that makes migrating away harder? Again, just random threads on Reddit. I feel like that with NextJS I can easily migrate, get back up and running, and if I need to migrate from NextJS to Astro that seems like a possibility. 

My two (fear-driven and illogical) cents.

**Reply by rbmedia** [reply]
I fear to experience with Astro the same I experienced with Gatsby: just to have  a very comfortable tool until it goes wrong (my opinion). 
I am back to plain React Apps, server components and custom data-API to serve content from a CMS the way I need it.

**Reply by robclancy** [reply]
Astro is great. Nextjs is a dumpsterfire of corporate bullshit. 

Also Astro's components are just whatever ones you want to use, react, svelte, whatever. The way component libraries should be. So not sure how that causes issues with migrating.  

To me the choice is more about use case. Do you want it fast and the option to have a real static site or to use something other than react? Astro. 
Do you want to essentially do what used to be done with rails, php etc but with frontend like code? Or do you want something with a huge community? Nextjs.

**Comment by rbmedia** [comment]
I stopped using Gatsby 3 months ago. Netlify is neither an option for future projects. Transparency and stability is what I need. The low activity on issues speak for itself...

**Reply by munawar-js** [reply]
We also stopped, now building new template as alternate for [gatsby-starter-foundation](https://github.com/stackrole/gatsby-starter-foundation) with Astro

**Reply by rbmedia** [reply]
I will not adapt any "in-between" development solution anymore, but use the main JS frameworks. Maybe I am wrong...

**Reply by munawar-js** [reply]
What you are saying is correct, initially i considered next.js which is my goto framework. But for static blog astro does the job

**Comment by JWizerd** [comment]
I have to say, in my experience as a dev over the past decade, I've never seen a situation like this so poorly handled. Netlify seems to have essentially ghosted an entire community to increase profits and cut out competition. I think they are failing to see that Gatsby, being a relatively stable and well architected framework, would probably be easy to maintain and could open up new profit opportunities like rolling out proper functioning incremental build pipelines which therefore would given incentive for existing customers to upgrade to enterpise tiers on their platform. But, in the process of boosting their profit margins they are going to be losing customers to other frameworks and cloud hosting platforms i.e. Next.js, Vercel, Hydrogen & Oxygen (Shopify). 

I hope this message reaches you Netlify and lights a fire under your a$$ to pick up support for this framework because just ghosting a community is not a good look for your image. At least have the courage to make an official statement that you're not going to be supporting the framework in the future... cowards.

**Reply by cordial** [reply]
well said.

**Reply by olehpratsko** [reply]
I feel exactly the same. What a shame that the stupid younger brother of Gatsby (Next.js) hyped the market like Gatsby was never there.

**Comment by olehpratsko** [comment]
```
For those migrating away, what's the preferred equivalent framework? NextJS? Nuxt? or?
```
I have more than 5 years of experience in Gatsby and some of my current projects are still working on Gatsby as a base framework (ex. https://minify-js.com/). Working with Gatsby was a great journey, I completely forgot about many painful points I struggled with in the previous life before meeting Gatsby. Hovever luck of support and demand on the webdev market, made me considering other options.

Let me start with **Astro**:

While it supports React components, Astro is not React based framework at all. It has it's own syntax (.astro), the syntax also looked much like Vue. It also reminded me about the old Jekyll, it actually looks like Jekyll written with JS. It might be easier in setting up static blog or some simple few pages website, but the general approach is much different from what we used to working with Gatsby.
I started questioning myself, why should I learn new no-name framework like Astro, if I can learn something like Vue-based Nuxt with great community and support, so I dropped the idea picking up Astro and moved to the more obvious Next.js.

Then I had 1 year of enterprise development with **Next.js**:
It was quick to pick up. Familiar ecosystem, modules, directory structure, but the devil hides in details. Next.js is an SSR framework and with introduction of server components, I'd say SSR is a primary way to build Next.js applications. While SSG is possible, Next.js has one important feature missing to make SSG working as good as in Gatsby. It doesn't have any native Data Layer. At the beggining I was straggling a lot. Working with Gatsby I completely forgot about doing direct requests to API endpoints, debugging API responses and other frustrating things related to the data fetching. Next.js reminded me everything and it really feels like a few steps back in my personal development flow. I couldn't get used to the new reality of exactly the same React ecosystem with pure data fetching, so at this point I don't consider Next.js as framework for my future projects. Of course, your experience could be different.

I started my personal portfolio project one month ago to summarise my 10 years in the industry and find some new opportunities and I got back to Gatsby for this project. For me personally, Gatsby works better then Next.js or Astro, but I also started exploring Nuxt.js as an alternative to Gatsby. I'm surprised, but it looks like Nuxt.js is very similar to Gatsby in the core concepts and as a big plus for me Nuxt.js is very opinionated. It handles lots of problems OOB, and for the problems it doesn't handle, it has external modules, just like Gatsby+plugins. It has connection modules for different CMS and service providers, just like Gatsby+source. It's based on Vue, but I don't see any not-next.js option based on React to replace Gatsby.

Well, I'd say I'll stick with Gatsby till the moment I'd be confident enough in development with Nuxt.js.

**Reply by roNn23** [reply]
That's a good summary and I can totally relate to it, or rather feel it. I haven't tried Nuxt.js, but all my other experiences have been the same. It's a shame what's happening, but I'm also sticking with Gatsby for now because it's still working great at the moment - but you can see the setting sun on the horizon and I'm a bit scared of the night that's coming.... 🌅

But there is a new release, so maybe everything is fine 🙏
https://github.com/gatsbyjs/gatsby/releases/tag/gatsby%405.14.0

**Reply by olehpratsko** [reply]
@roNn23 Unfortunately this release is nothing major and only confirms the current Gatsby state as sunsetting. It includes very minor updates:
- Minor docs updates, 3 of 4 are dead links updates.
- Automatic dependabot updates.
- And finally, removing telemetry from the utils because the telemetry service was shutdown. It also indicates that they are not interested in collecting Gatsby usage data, which is very bad sign for us.

**Reply by roNn23** [reply]
Thank you for pointing that out. Nevertheless, I won't bury my hopes that Gatsby will experience another rebirth or at least maintain its current status. We will see.

**Comment by serhalp** [comment]
Hi folks.

For those who haven't seen it, we responded to these questions about a year ago in [this comment](https://github.com/gatsbyjs/gatsby/issues/38696#issuecomment-1817064739) and [this blog post](https://www.netlify.com/blog/developing-gatsby-enhancing-stability-and-reliability/).

Nothing has changed since and we have no plans to change this approach.

I believe the current level of limited development (security fixes, limited dependency updates, low-hanging-fruit bug fixes and perf improvements, addressing upcoming third-party API deprecations, etc.) is aligned with what we've communicated. If you feel otherwise, please let us know how can we do better.

Thanks!

**Reply by rbmedia** [reply]
Hi, 
thanks for your explanation, may you read the comments below the one you linked, from a year ago, it does not sounds different than what we discussed here. 

When I read the sentence: "It is no secret that Gatsby is no longer the hip new framework on the block", I have to say that I was using Gatsby in production, thinking in a long term solution, never looking for a hipe framework, I stopped using it now.
I am nearer to Vercel than to Netlify.

**Comment by JWizerd** [comment]
@serhalp Your comment https://github.com/gatsbyjs/gatsby/issues/38696#issuecomment-1817064739 says you intend on supporting the Gatsby framework. Ok, prove it then... it's been over a year since your team has made any meaningful contribution. Furthermore, I reached out to Netlify support directly to get answers about future support of Gatsby who put me in touch with Elad Rosenheim, a project manager at Netlify, who stated that as far as Gatsby the open source framework is considered, it would most likely NOT be supported anymore.

Your comment also indicates that Gatsby will be a "safe, robust and reliable choice to build production quality websites and e-commerce stores". Therefore, I feel the need to elevate, as well as help out the community by bringing this to your attention... Because of the lack of the continued support of the Gatsby framework, and lack of support for gatsby plugins as well, in Jan 2025, the gatsby-source-shopify plugin (https://www.gatsbyjs.com/plugins/gatsby-source-shopify/) is going to break because the API version 2024-01 will be sunsetted and will be forwarded to the next stable version which has major breaking changes to the API that have not been reflected in the latest version of the plugin. This means any merchants using this plugin will no longer be able to build their site in 2025. Currently, there are 10k+ weekly downloads on that plugin on NPM. This is the opposite of a safe and reliable choice for eCommerce websites wouldn't you say?

Frankly, I've lost trust in your team to effectively support this framework, and my company has opted for migrating over to Hydrogen rather than risk waiting for you to allocate resources to supporting this framework. But, I felt the need to communicate the risks of sticking with Gatsby, especially Shopify merchants, to help other people in the community. I wish you all the best of luck!

**Reply by roNn23** [reply]
@JWizerd Good point. I'll have a look at [Hydrogen](https://hydrogen.shopify.dev/) as a Gatsby replacement. Thanks for that.
@serhalp Will you work on those Plugins? I have not used the plugin, but how exactly do you deal with this problem in the future? Is there a roadmap for this? Thank you.

**Reply by serhalp** [reply]
> in Jan 2025, the gatsby-source-shopify plugin (https://www.gatsbyjs.com/plugins/gatsby-source-shopify/) is going to break because the API version 2024-01 will be sunsetted and will be forwarded to the next stable version which has major breaking changes to the API that have not been reflected in the latest version of the plugin. This means any merchants using this plugin will no longer be able to build their site in 2025. Currently, there are 10k+ weekly downloads on that plugin on NPM. This is the opposite of a safe and reliable choice for eCommerce websites wouldn't you say?

To address this question specifically — we will be updating gatsby-source-shopify to the latest Shopify API version before the current API version's EOL (January 1). We [addressed this here](https://github.com/gatsbyjs/gatsby/discussions/39129).

**Comment by olehpratsko** [comment]
@serhalp 
Yes, we all have seen the comment from @biilmann.
```
Gatsby is here to stay and we’re not sunsetting it. We’re committed to not only maintain it but also to building a forward-looking roadmap for it as we go into 2024, which I’ll share with you all as soon as we can.
```
We are going into 2025 now, but 2024 roadmap for Gatsby was never shared. I was checking Netlify and Gatsby blogs and I couldn't find it. Please, correct me if I am wrong or it were published anywhere else.

I agree with @JWizerd comment above. With lack of support of the core and plugins Gatsby is not reliable solution for production websites.

**Reply by sloyer** [reply]
5 months into 2025, no roadmap.

**Reply by Christopher96u** [reply]
7 months into 2025, no roadmap.

**Reply by brennanbrown** [reply]
10 months into 2025. no roadmap.

**Reply by Kidkie** [reply]
12 months into 2025, no roadmap.

**Reply by Christopher96u** [reply]
15 months into 2026, no roadmap. What a joke

**Comment by joebentaylor1995** [comment]
Gatsby has been dead the second Netlify bought the product. Ill never forgive you Kyle.

**Comment by muditjuneja** [comment]
Damn! It is really dead man. I am sad, I have to migrate my 2 projects to an alternate stack :/ :/ :/

**Reply by ZeldOcarina** [reply]
Yeah go Astro, it's so good!

**Comment by OloZ17** [comment]
Too much time without any clear response regarding GatsbyJS, I just migrated 4 sites to Next.js. It's time-consuming to do, but I prefer to invest my time now rather than lose it later.

**Reply by cordial** [reply]
Did you migrate to next.js with it being fully SSG?

**Reply by ZeldOcarina** [reply]
Astro is a better replacement for what Gatsby was doing..

**Reply by richard-underwood** [reply]
React Router (framework mode - what was Remix) in SSG mode works well. What is missing (from this, nextJS or Astro) is the graphql/content layer for retrieving/manipulating content (particularly images) at dev/build time, which was really one of the key GatsbyJS features. If anyone has a suggestion for a replacement for that, it'd be appreciated.

**Reply by robclancy** [reply]
Astro content layer is similar. https://astro.build/blog/content-layer-deep-dive/

It's had a bunch of updates since that post too.

**Reply by OloZ17** [reply]
> Did you migrate to next.js with it being fully SSG?

Yeah.

**Reply by OloZ17** [reply]
But the best part, i discovered with Next.js => next-safe-action

**Reply by cordial** [reply]
> > Did you migrate to next.js with it being fully SSG?
> 
> Yeah.

did u find any good resource to do it or did you just take it one page/template at a time?

**Reply by cordial** [reply]
> But the best part, i discovered with Next.js => next-safe-action

interesting, cheers.

**Comment by zlterry69** [comment]
So… are we just pretending Gatsby isn’t dead?
No roadmap, no React 19 support, half the plugins broken, and total silence from Netlify. It’s like watching a once-great framework slowly decompose while everyone acts like it’s fine.

If this is “maintenance mode,” at least have the guts to say it so devs can move on instead of waiting for a miracle that’s clearly not coming.

**Comment by darkav** [comment]
I used to love Gatsby, but it’s clearly abandoned now. No updates, no roadmap, and everything’s breaking slowly. Sad to see such a great project end like this.

**Comment by tatan22** [comment]
Tried upgrading Gatsby from v4 to v5 and it’s a nightmare.
Half the plugins don’t even load, GraphQL queries break in weird ways, and the migration guide is basically a joke. Feels like we’re beta-testing on production with zero support.

We were promised a clear roadmap of what’s going on with the framework, and after more than a year, there is still no one saying anything!!! It’s frustrating that issues sit for months — even years — without any response from maintainers. Sad to say, but Gatsby was fucked up since was acquired by Netlify.

**Reply by ZeldOcarina** [reply]
Yeah Astro is the way nowadays!!

**Comment by CodyEngel** [comment]
Came here to pour one out for Gatsby. I hardly knew ye but seemed like you were a great friend right up until you learned about capitalism.
